core/src/main/scala/cats/data/NonEmptySet.scala

```diff
@@ -428,7 +428,7 @@ sealed abstract private[data] class NonEmptySetInstances extends NonEmptySetInst
 }
 
 sealed abstract private[data] class NonEmptySetInstances0 extends NonEmptySetInstances1 {
-  implicit def catsDataHashForNonEmptySet[A: Order: Hash]: Hash[NonEmptySet[A]] =
+  implicit def catsDataHashForNonEmptySet[A: Hash]: Hash[NonEmptySet[A]] =
```

Comment:
Can someone explain (or guess at least) why does this change not concern Mima? I removed the unused implicit `Order` parameter but seems that Mima is completely fine about that.